Mathlib/Data/Fin/Tuple/Sort.lean

```diff
@@ -107,6 +109,103 @@ theorem monotone_sort (f : Fin n → α) : Monotone (f ∘ sort f) := by
   exact (monotone_proj f).comp (graphEquiv₂ f).monotone
 #align tuple.monotone_sort Tuple.monotone_sort
 
+/-- All the elements `· ≤ a` appear the start of a sorted tuple -/
+theorem sort_lt_at_start_of_monotone {α} [LinearOrder α] (m : ℕ) (f : Fin m → α) (a : α)
+    (h_sorted : Monotone f)
+    (j : Fin m) (h : j < Fintype.card {i // f i ≤ a}) :
+    f j ≤ a := by
+  contrapose! h
+  rw [← Fin.card_Iio, Fintype.card_subtype]
+  refine Finset.card_mono (fun i => Function.mtr ?_)
+  simp_rw [Finset.mem_filter, Finset.mem_univ, true_and, Finset.mem_Iio]
+  exact fun hij => (h.trans_le <| h_sorted <| le_of_not_lt hij).not_le
+/- Proofs by Ruben Van de Velde, Eric {Rodriguez and Wieser}, Junyan Xu -/
```

Comment:
```suggestion
```
This probably shouldn't be in the source code; this type of thing is best left to the PR description, perhaps via `Co-authored-by`.